### PR TITLE
fix(maxmsp): Global(indexed access), Dict(contructor), Maxobj(patchcords)

### PR DESCRIPTION
**Pull request by FrancescElies**
The [Global Object](https://docs.cycling74.com/max8/vignettes/jsglobalobject) is used to set your own properties
but as it's currently defined typescript will immediately complain because they are not defined in the class.

The current solution is to cast the global to any, which defeats the whole purpose of using types (see test example).

This PR allows an indexer and forces the user to do type narrowing and make sure at runtime that he has the type he expects.

### CODE REVIEW

File: types/maxmsp/index.d.ts
Comment (FrancescElies) [inline types/maxmsp/index.d.ts:571]: We could use `any` instead o `unkown` as it has been done in many other places in this definitions, but I tend to think any is too loose.

Comment (twhiston) [reply types/maxmsp/index.d.ts:571]: isn't this more of a pain when we need to get the value though, as you need to then explicitly cast it, eg to a `string`, which you don't need with `any`

File: types/maxmsp/maxmsp-tests.ts
Comment (FrancescElies) [inline types/maxmsp/maxmsp-tests.ts:50]: I think this is a better way of approaching this and doen't require to change the Global class definition.

In the previous example anything under `glbl` is `any` leaving the door open whatever might happen at runtime

Comment (FrancescElies) [reply types/maxmsp/maxmsp-tests.ts:50]: Tdaoy I gave it a test to this internally and I keep going back and forth, I am tending to think the previous approach was better, having to subclass just to get around typescript is not really good either.

File: types/maxmsp/index.d.ts
Comment (FrancescElies) [reply types/maxmsp/index.d.ts:589]: I changed my mind, I updated the tests, I still believe the indexer approach is better than having to cast global to `any` before it can be used (in that case one can use ts-ignore which would have the same effect).

File: types/maxmsp/maxmsp-tests.ts
Comment (DangerBotOSS) [inline types/maxmsp/maxmsp-tests.ts:55]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by twhiston — CHANGES_REQUESTED** [review]
Can we split this into 2 PR's please? All the non-global stuff is an easy approve, but I'm really not sure about the Global topic. I don't like having to subclass it and it will break existing code I think, so I'd like to have that in it's own discussion where we can think about it further and I can test that change alone against some of my existing code. Thanks!

**Comment by FrancescElies** [comment]
@twhiston I updated `types/maxmsp/maxmsp-tests.ts` with the three ways I see one could work with Global.

Unless I am missing something, the old way of doing things still should work.

**Comment by twhiston** [comment]
Sorry will try to check this over the weekend or early next week, I've been ill this week so not been working.

**Comment by FrancescElies** [comment]
No rush at all, i am not in need of this being merged :)

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Marking this as request change for now.
Please reopen or update when this is ready for review

**Comment by FrancescElies** [comment]
@sheetalkamat this is ready for review, let’s please wait for @twhiston

**Review by twhiston — APPROVED** [review]
Firstly apologies this took me so long to look at. I've had a proper look through this now and I think it's fine to be merged. It won't break any existing code as you can still cast to `<any>` as per `src/ExampleJS.ts` in https://github.com/ErnstHot/TypeScript-for-Max but you can also choose to be safer by using type guards or a subclass.

Therefore I'd leave all three of your examples in the test file, to show people various ways to work with this and leave it up to user choice. Sound good to you?

**Comment by FrancescElies** [comment]
@sheetalkamat could you please allow us to merge this

**Comment by twhiston** [comment]
I don't see there's an actual merge conflict here 😕 maybe we just need @sheetalkamat to approve as well? If not you could close this PR completely and open a new one, that would allow us to progress with the normal workflow

**Comment by FrancescElies** [comment]
superseeded by #66857

**Pull request closed without merging**
